src/ecalc/cli/tests/test_app.py

```diff
@@ -301,6 +301,46 @@ def test_json_true_detailed_output(self, simple_yaml_path, tmp_path, snapshot):
             json.dumps(json_data, sort_keys=True, indent=2, default=str), snapshot_name=v3_json_actual_path.name
         )
 
+    def test_json_advanced_model(self, advanced_yaml_path, tmp_path):
+        """Check advanced json file to ensure compressor requested inlet- and outlet
+        pressures are reported correctly.
+
+        :param advanced_yaml_path: eCalc model with some detailed compressor setup
+        :param tmp_path: temporary json results
+        :return: nothing
+        :raises: AssertionError if eCalc reports wrong requested inlet- and outlet pressures.
+        """
+        run_name_prefix = "test_json_advanced_model"
+        runner.invoke(
+            main.app,
+            _get_args(
+                model_file=advanced_yaml_path,
+                json=True,
+                output_folder=tmp_path,
+                name_prefix=run_name_prefix,
+                detailed_output=True,
+            ),
+            catch_exceptions=False,
+        )
+
+        v3_json_actual_path = tmp_path / f"{run_name_prefix}_v3.json"
+        assert v3_json_actual_path.is_file()
+        with open(v3_json_actual_path) as json_file:
+            json_data = json.loads(json_file.read())
+
+        compressor_tabular = json_data["models"][0]
+        compressor_train = json_data["models"][1]
+
+        requested_inlet_pressure_tabular = compressor_tabular["requested_inlet_pressure"]["values"]
+        requested_outlet_pressure_tabular = compressor_tabular["requested_outlet_pressure"]["values"]
+        requested_inlet_pressure_train = compressor_train["requested_inlet_pressure"]["values"]
+        requested_outlet_pressure_train = compressor_train["requested_outlet_pressure"]["values"]
+
+        assert requested_inlet_pressure_tabular == [20] * len(compressor_tabular["timesteps"])
```

Comment:
do we have the calculated pressure here as well? does it differ from the requested?

Reply:
I cannot find it in the json data object, read from disk. Seems like inlet_stream and outlet_stream (where calculated pressures are) are not written to json. Even though it is part of the dto that goes in to the write_json function:
![image](https://github.com/equinor/ecalc/assets/70878501/2d093cd9-9cdc-4a93-9a0d-4836b3ba5df1)

Reply:
@TeeeJay any ideas why inlet_stream is not written to json, is it because of the CompressorStreamCondition class - mapping issue?

Reply:
Found the calculated pressures in stage results.